### PR TITLE
feat: automatically add all existing users as owners to all existing …

### PR DESCRIPTION
**Pull request by chriswk**
…projects

### CODE REVIEW

File: src/migrations/20210428103923-onboard-projects-to-rbac.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                    ((SELECT role_id from project_member), '${p.name}', 'UPDATE_CONTEXT_FIELD'),
+                    ((SELECT role_id from project_member), '${p.name}', 'DELETE_CONTEXT_FIELD'),
+                    ((SELECT role_id from project_member), '${p.name}', 'CREATE_FEATURE'),
+                    ((SELECT role_id from project_member), '${p.name}', 'UPDATE_FEATURE'),
+                    ((SELECT role_id from project_member), '${p.name}', 'DELETE_FEATURE');
+
+                WITH owner_id AS (
+                    SELECT id FROM roles WHERE type='project' AND project='${p.name}' AND name = 'Owner'
+                )
+                INSERT INTO role_user(role_id, user_id) SELECT o.id, u.id FROM owner_id o, users u ON CONFLICT DO NOTHING;
```
Comment (ivarconr) [inline src/migrations/20210428103923-onboard-projects-to-rbac.js:47]: this was quite interesting. have nit seen this FROM syntax before

File: src/migrations/20210428103923-onboard-projects-to-rbac.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                        RETURNING id role_id
+                )
+                INSERT INTO role_permission(role_id, project, permission) VALUES
+                    ((SELECT role_id from project_member), '${p.name}', 'CREATE_STRATEGY'),
+                    ((SELECT role_id from project_member), '${p.name}', 'UPDATE_STRATEGY'),
+                    ((SELECT role_id from project_member), '${p.name}', 'DELETE_STRATEGY'),
+                    ((SELECT role_id from project_member), '${p.name}', 'UPDATE_APPLICATION'),
+                    ((SELECT role_id from project_member), '${p.name}', 'CREATE_CONTEXT_FIELD'),
+                    ((SELECT role_id from project_member), '${p.name}', 'UPDATE_CONTEXT_FIELD'),
+                    ((SELECT role_id from project_member), '${p.name}', 'DELETE_CONTEXT_FIELD'),
```
Comment (ivarconr) [inline src/migrations/20210428103923-onboard-projects-to-rbac.js:39]: These are actually not project permissions. Only feature toggles lives under a project.

File: src/migrations/20210428103923-onboard-projects-to-rbac.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                    db.runSql.bind(
+                        db,
+                        `
+                WITH project_owner AS (
+                    INSERT into roles (name, description, type, project)
+                        VALUES ('Owner', '${DESCRIPTION.OWNER}', 'project', '${p.name}')
+                        RETURNING id role_id
+                )
+                INSERT INTO role_permission(role_id, project, permission)
+                SELECT role_id, '${p.name}', 'ADMIN' from project_owner;
```
Comment (ivarconr) [inline src/migrations/20210428103923-onboard-projects-to-rbac.js:25]: The project owner role needs the following permissions:

- UPDATE_PROJECT
- DELETE_PROJECT
- CREATE_FEATURE
- UPDATE_FEATURE
- DELETE_FEATURE

File: src/migrations/20210428103923-onboard-projects-to-rbac.js (hunk 2 above)
Comment (chriswk) [reply src/migrations/20210428103923-onboard-projects-to-rbac.js:39]: Good point. I'll fix :)

File: src/migrations/20210428103923-onboard-projects-to-rbac.js (hunk 3 above)
Comment (chriswk) [reply src/migrations/20210428103923-onboard-projects-to-rbac.js:25]: Will fix

File: src/migrations/20210428103923-onboard-projects-to-rbac.js (hunk 2 above)
Comment (chriswk) [reply src/migrations/20210428103923-onboard-projects-to-rbac.js]: fixed in caa7225352fcd4469a138fa05575271e35aa8686

File: src/migrations/20210428103923-onboard-projects-to-rbac.js (hunk 3 above)
Comment (chriswk) [reply src/migrations/20210428103923-onboard-projects-to-rbac.js]: fixed in caa7225352fcd4469a138fa05575271e35aa8686

**Review by ivarconr — APPROVED** [review]

**Pull request merged**